lib/saml_idp/xml_security.rb

```diff
@@ -196,38 +226,39 @@ def verify_signature(base64_cert, sig_alg, signature, canon_string, soft)
         signature_algorithm = algorithm(sig_alg)
 
         unless cert.public_key.verify(signature_algorithm.new, signature, canon_string)
-          return soft ? false : (raise ValidationError.new("Key validation error"))
+          return soft ? false : (raise ValidationError.new('Key validation error'))
         end
 
-        return true
+        true
       end
 
       def digests_match?(hash, digest_value)
         hash == digest_value
       end
 
       def extract_signed_element_id
-        reference_element       = REXML::XPath.first(self, "//ds:Signature/ds:SignedInfo/ds:Reference", {"ds"=>DSIG})
-        self.signed_element_id  = reference_element.attribute("URI").value[1..-1] unless reference_element.nil?
+        reference_element = REXML::XPath.first(self,
+                                               '//ds:Signature/ds:SignedInfo/ds:Reference', { 'ds' => DSIG })
+        return if reference_element.nil?
+
+        self.signed_element_id = reference_element.attribute('URI').value[1..-1]
       end
 
       def canon_algorithm(element)
         algorithm = element.attribute('Algorithm').value if element
         case algorithm
-        when "http://www.w3.org/2001/10/xml-exc-c14n#"         then Nokogiri::XML::XML_C14N_EXCLUSIVE_1_0
-        when "http://www.w3.org/TR/2001/REC-xml-c14n-20010315" then Nokogiri::XML::XML_C14N_1_0
-        when "http://www.w3.org/2006/12/xml-c14n11"            then Nokogiri::XML::XML_C14N_1_1
+        when 'http://www.w3.org/2001/10/xml-exc-c14n#'         then Nokogiri::XML::XML_C14N_EXCLUSIVE_1_0
+        when 'http://www.w3.org/TR/2001/REC-xml-c14n-20010315' then Nokogiri::XML::XML_C14N_1_0
+        when 'http://www.w3.org/2006/12/xml-c14n11'            then Nokogiri::XML::XML_C14N_1_1
         else                                                        Nokogiri::XML::XML_C14N_EXCLUSIVE_1_0
```

Comment:
I don't like `then` at all IMO i'd just remove it
```suggestion
        when 'http://www.w3.org/2001/10/xml-exc-c14n#'
          Nokogiri::XML::XML_C14N_EXCLUSIVE_1_0
        when 'http://www.w3.org/TR/2001/REC-xml-c14n-20010315'
          Nokogiri::XML::XML_C14N_1_0
        when 'http://www.w3.org/2006/12/xml-c14n11'
          Nokogiri::XML::XML_C14N_1_1
        else
          Nokogiri::XML::XML_C14N_EXCLUSIVE_1_0
```

Reply:
fixed in https://github.com/18F/saml_idp/pull/98/commits/2b8c1afa357e64a05ddf8932dfae209da2a40a6a